### PR TITLE
Release 22.12.0

### PR DESCRIPTION
**Pull request by adiroiban**
# Description

This is the release branch for 22.12.0

### CODE REVIEW

File: pyproject.toml
Comment (adiroiban) [inline pyproject.toml:84]: The incremental related tests was failing for the release candidate version format...so at first I try updating incremental with the hope that normalization is fixed.

It is not... but maybe is ok to ask for a newer incremental.

Best, would be to not depend on incremental :)

Let me know if this should be reverted or moved to a separate PR.

File: src/towncrier/_project.py
Comment (adiroiban) [inline src/towncrier/_project.py:16]: I renamed this as the generic `Version` was a bit confusing

File: src/towncrier/test/test_project.py
Comment (adiroiban) [inline src/towncrier/test/test_project.py:58]: This is the failing test for RC.

I renamed to `metadata_version` as the generic `version` was a bit confusing.

File: NEWS.rst
Comment (hynek) [inline NEWS.rst:19]: we should update the template – all PR links still point to Hawkie

File: src/towncrier/_project.py
Comment (adiroiban) [inline src/towncrier/_project.py:54]: Not sure if this hack is ok... I am not happy with it.. but at the same time I don't know what to do.

I would rather have a green test, than remove the test.

**Review by adiroiban — COMMENTED** [review]
This should be ready for review

**Comment by adiroiban** [comment]
The release candidate and tag have not yet been created in PyPI and I am waiting for an initial review of this PR.
I am not sure if the changes are ok... they result in green tests, but might not be the best way to fix the tests.

Thanks!

**Review by hynek — APPROVED** [review]
yolo

**Comment by adiroiban** [comment]
Thanks for the review. I will try to follow the process from [here](https://towncrier.readthedocs.io/en/latest/release.html) and do a public pre-release on PyPI

**Comment by adiroiban** [comment]
release candidate triggered

* here https://github.com/twisted/towncrier/releases/tag/22.12.0rc1
* and here https://github.com/twisted/towncrier/actions/runs/3741354542

If all is OK I will update the version and do the final release.

**Comment by webknjaz** [comment]
@adiroiban hey, I noticed https://github.com/twisted/towncrier/releases/tag/22.12.0rc1.

May I suggest you to use the `☑️ Create a discussion for this release` checkbox when making the final release? This allows having discussions linked to GH Releases, creating a dedicated space for people to share a release feedback in a consolidated place. Here's an example of what it looks like: https://github.com/ansible-community/ansible-test-gh-action/releases/tag/v1.12.0 — there's a `💬 Join discussion` button in the bottom right corner that people may follow.

P.S. This probably requires enabling Discussions in the repo settings before making the release. In older repos, it is often needed to also create a separate category, typically called `📢 Announcements` and using the `🔒 Announcement` discussion format. I also typically type `Updates from maintainers` into the description field of the category.

---

Another observation is that you pasted RST there, but it only supports Markdown. In https://github.com/twisted/towncrier/issues/353#issuecomment-1359432962, I've linked to a GHA-based example, but a short TL;DR version for local invocations would be:
```shell
sudo apt install -y pandoc && pandoc --from=rst --to=gfm --output=changelog.md changelog.rst
```

N.B. `gfm` stands for GitHub-Flavored Markdown.

**Comment by webknjaz** [comment]
> and here [twisted/towncrier/actions/runs/3741354542](https://github.com/twisted/towncrier/actions/runs/3741354542)

If you look at https://github.com/twisted/towncrier/actions/runs/3741354542/jobs/6350891579#step:9:19 closer, you'll see warnings being emitted:
```console
Checking dist/towncrier-22.12.0rc1-py3-none-any.whl: PASSED with warnings
WARNING  `long_description_content_type` missing. defaulting to `text/x-rst`.
Checking dist/towncrier-22.12.0rc1.tar.gz: PASSED with warnings
WARNING  `long_description_content_type` missing. defaulting to `text/x-rst`.
```

P.S. In general, it's a good idea to run `twine check --strict` at early stages. My pypi-publish action also has a strict mode, but it's not pleasant to see hard failures at the very last moment before the release.

**Comment by adiroiban** [comment]
PyPi release candidate is live https://pypi.org/project/towncrier/22.12.0rc1/

----------

Thanks @webknjaz  for the input

I was expecting that we already run something like `twine check --strict` :(
We should do it... in a separate PR :)

--------

This is what I have when creating a GitHub release

I don't see the create discussion button... I think it needs the "discussion" feature.

**Comment by webknjaz** [comment]
> I don't see the create discussion button... I think it needs the "discussion" feature.

It'll be a drop-down when discussions are enabled. Note that you can only do this on GH Release creation, updating an existing one won't work.

**Comment by webknjaz** [comment]
Happy to report that the pre-release doesn't break the smoke tests of my Sphinx extension: https://github.com/sphinx-contrib/sphinxcontrib-towncrier/actions/runs/3742185592.

**Comment by adiroiban** [comment]
We don't close the comments on a PR. I think we can use the release PR for discussion.
There is not much activity over town crier. I think the discussions for the PRs and issues are more that we need.

I don't have much time for towncrier.
I think the current release notes from GitHub releases are good enough. Yes...the links are not nice...but we should be fine.  

In theory with the new --keep feature we should be able to have 2 templates...one for RST and one for Markdown

When possible I try  not to sell my soul to platform specific features 😀

File: NEWS.rst
Comment (pawelad) [inline NEWS.rst]: I believe there's a typo here - `22.22.0` instead of `22.12.0`.

Comment (adiroiban) [reply NEWS.rst]: Thanks. Much appreciated

**Comment by adiroiban** [comment]
22.10.0 GHA release triggered [here](https://github.com/twisted/towncrier/actions/runs/3750164167)

[PyPI is ready](https://pypi.org/project/towncrier/)

----------

Many thanks for your help with this release.

---------

Just a note. It's hard for me to find free time for doing/driving the towncrier release.
If anyone want to help with driving / pushing new towncrier releases, the release process is documented here https://towncrier.readthedocs.io/en/stable/release.html

I would be happy to guide and assist

**Comment by webknjaz** [comment]
> We don't close the comments on a PR. I think we can use the release PR for discussion. There is not much activity over town crier. I think the discussions for the PRs and issues are more that we need.

Many folks don't like leaving comments on closed things. Besides, issues/PRs are more about bugs then feedbacks. This PR diff does not include all the changes that got into the release anyway. Discussions have a different structure and semantic purpose. Many maintainers experimented sending folks to Discussions for community support to keep the bug tracker filled with things they plan to work on. Here's one example of how it's done: https://github.com/pradyunsg/furo/issues/new/choose.

> I don't have much time for towncrier. I think the current release notes from GitHub releases are good enough. Yes...the links are not nice...but we should be fine.
> 
> In theory with the new --keep feature we should be able to have 2 templates...one for RST and one for Markdown

That wouldn't work well because the fragments may use bits of specific syntax. So if people are enclosing RST fragments, and they are injected into the changelog via Jinja without any sort of conversion, you'll get a “Markdown” file filled with RST syntax all over the place and that's exactly what we see now.

> When possible I try not to sell my soul to platform specific features :grinning:

True, but PRs are also platform-specific to platforms. They aren't part of Git :-P
I like how discussions provide a place for people to interact with each other without disturbing the maintainers.

**Comment by webknjaz** [comment]
> If anyone want to help with driving / pushing new towncrier releases, the release process is documented here [towncrier.readthedocs.io/en/stable/release.html](https://towncrier.readthedocs.io/en/stable/release.html)

I can look at automating the process more sometime, maybe, if that'll help.

**Comment by adiroiban** [comment]
**Discussions** are now back online.

I love automation :) so, bring it in :)

**Pull request merged**
